### PR TITLE
Tests factories and ex machina

### PR DESCRIPTION
**Pull request by flmel**
### Description
**work in progress**
Added ex_machina in order to DRY tests did rework the accounts tests
### Issue
#316 and #60 

### Screenshots

_For frontend updates, please include a screenshot._

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: test/chat_api/accounts_test.exs [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
 
   describe "accounts" do
     alias ChatApi.Accounts.Account
 
-    @valid_attrs %{company_name: "some company_name"}
-    @update_attrs %{company_name: "some updated company_name"}
-    @invalid_attrs %{company_name: nil}
+    @valid_attrs params_for(:account)
+    @update_attrs params_for(:account, company_name: "updated company name")
+    @invalid_attrs params_for(:invalid_account)
```
Comment (reichert621) [inline test/chat_api/accounts_test.exs:12]: would this work if we did `params_for(:account, company_name: nil)`?

I'm not sure if we actually need to define fixtures for every "invalid" type 🤔

**Comment by reichert621** [comment]
looks reasonable so far @flmel! thanks :)

File: test/chat_api/accounts_test.exs (hunk 1 above)
Comment (flmel) [reply test/chat_api/accounts_test.exs:12]: invalid_attrs params_for(:account, "")  would work ofcourse nil value would skip the field

**Comment by reichert621** [comment]
looks great so far @flmel! it's quite long so it might take me some to review all of it 😅 but looking forward to merging your updates 🎉

**Comment by flmel** [comment]
> looks great so far @flmel! it's quite long so it might take me some to review all of it 😅 but looking forward to merging your updates 🎉

the only issue would be is that it blocks further backend tasks since adding more tests would cause catch22

**Comment by reichert621** [comment]
yup good point -- sorry about that! i'll make sure we prioritize this before merging any new backend tests 👍

**Review by reichert621 — COMMENTED** [review]
amazing work on this @flmel! 🚀  🚀  🚀 

simplifies tests, and also speeds them up 😄 (i think it went from ~30s before to now around ~10s on my laptop!)

**Pull request merged**